### PR TITLE
gowin: Use dfflegalize.

### PR DESCRIPTION
**Pull request by mwkmwkmwk**

### CODE REVIEW

**Comment by pepijndevos** [comment]
This looks like it makes stuff a lot simpler, but I have no idea how `dfflegalise` works, so need to do some study before I can meaningfully review this.

File: tests/arch/gowin/init.ys [hunk 1]
```diff
@@ -38,8 +38,8 @@ chparam -set INIT 1 myDFF myDFFN myDFFE myDFFNE
 
 # async should give a warning
```
Comment (pepijndevos) [inline tests/arch/gowin/init.ys:39]: This comment is now incorrect.

File: tests/arch/gowin/init.ys [hunk 2]
```diff
@@ -38,8 +38,8 @@ chparam -set INIT 1 myDFF myDFFN myDFFE myDFFNE
...
 # sync should synth to a mux
-chparam -set INIT 0 myDFF*S* myDFF*P*
-chparam -set INIT 1 myDFF*R* myDFF*C*
+chparam -set INIT 0 myDFF*S*
+chparam -set INIT 1 myDFF*R*
```
Comment (pepijndevos) [inline tests/arch/gowin/init.ys:42]: Can you explain the effect of removing the Preset/Clear async flops here? If I'm understanding the docs correctly, this should be handled by the `- emulating adff when the (reset, init) value combination is unsupported by
  dff + adff + dlatch + mux` case. In this case, I'd expect it to be reflected in changes to the following `-assert-count` lines. Or do these cases produce an error now? Is there a way to test that they do in fact result in an error?

Comment (mwkmwkmwk) [reply tests/arch/gowin/init.ys:42]: That would be true if `synth_gowin` actually supported latches — without them, it's an error.

Comment (pepijndevos) [reply tests/arch/gowin/init.ys:42]: Ah true. I guess I'll add those later. Should be fairly easy. I guess all I need to do once I add those is add a `-cell $_DLATCH_[NP]_ ???` directive to tell it these are supported.

Comment (mwkmwkmwk) [reply tests/arch/gowin/init.ys:42]: Depends, what capabilities do these latches have? If they're anything like FFs (ie. single s/r pin of const polarity, varying polarity enable, init tied to reset value), then `-cell $_DLATCH_?P?_ r` would be correct

File: tests/arch/gowin/init.ys (hunk 1 above)
Comment (mwkmwkmwk) [reply tests/arch/gowin/init.ys]: Removed

File: tests/arch/gowin/init.ys (hunk 2 above)
Comment (mwkmwkmwk) [reply tests/arch/gowin/init.ys:41]: I've added an error test for now.

**Pull request merged**